dataproc/quickstart/quickstart.py

```
@@ -0,0 +1,125 @@
+        },
+        'pyspark_job': {
+            'main_python_file_uri': job_file_path
+        }
+    }
+
+    job_response = job_client.submit_job(project_id, region, job)
+    job_id = job_response.reference.job_id
+
+    print('Submitted job \"{}\".'.format(job_id))
+
+    # Termimal states for a job.
+    terminal_states = {
+        dataproc.types.JobStatus.ERROR,
+        dataproc.types.JobStatus.CANCELLED,
+        dataproc.types.JobStatus.DONE
+    }
+
+    # Create a timeout such that the job gets cancelled if not in a
+    # terminal state after a fixed period of time.
+    timeout = 600000
+    time_start = time.time()
+
+    # Wait for the job to complete.
+    while job_response.status.state not in terminal_states:
+        if time.time() > time_start + timeout:
+            job_client.cancel_job(project_id, region, job_id)
+            print('Job {} timed out after threshold of {} seconds.'.format(
+                job_id, timeout / 1000))
+
+        # Poll for job termination once a second.
+        time.sleep(1)
+        job_response = job_client.get_job(project_id, region, job_id)
+
+    # Cloud Dataproc job output gets saved to a GCS bucket allocated to it.
+    cluster_info = cluster_client.get_cluster(project_id, region, cluster_name)
+
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(cluster_info.config.config_bucket)
+    output_blob = (
+        'google-cloud-dataproc-metainfo/{}/jobs/{}/driveroutput.000000000'
+        .format(cluster_info.cluster_uuid, job_id))
+    output = bucket.blob(output_blob).download_as_string()
+
+    print('Job {} finished with state {}:\n{}'.format(
+        job_id,
+        job_response.status.State.Name(job_response.status.state),
+        output))
+
+    # Delete the cluster once the job has terminated.
+    cluster_client.delete_cluster(project_id, region, cluster_name).result()
+
+    print('Cluster {} successfully deleted.'.format(cluster_name))
```

In the create_cluster_test it looked like cluster_client.delete_cluster was asynchronous and we had to wait for the operation to complete to verify success. Why not do that here? This print statement could fire and pretend to be a success even if the delete call eventually failed silently.

---

This is also an asynchronous call that blocks until completed, I just short-handed it into a single line. However, I'll split it into two lines as I did with `create_cluster` to make this a bit clearer.

---

Oh, whoops! I should have read it more carefully. But, sure, for clarity two lines here seems like a good idea.